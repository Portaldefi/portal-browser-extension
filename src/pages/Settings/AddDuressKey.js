import AddDuressKey from './AddDuressKey';
import { create, act } from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';

describe('AddDuressKeyComponent', () => {
  it('should work', async () => {
    let tree;
    act(() => {
      tree = create(
        <BrowserRouter>
          <AddDuressKey />
        </BrowserRouter>
      );
    });

    expect(tree).toMatchSnapshot();
  });
  afterAll(() => jest.resetModules());
});