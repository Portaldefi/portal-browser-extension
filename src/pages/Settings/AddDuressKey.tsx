import React, {useCallback, useState} from 'react';
import { useAppDispatch } from '@/hooks';
import { Grid, Header, Form, Button } from 'semantic-ui-react';
import { useNavigate } from 'react-router-dom';


export default () => {
    const [duressKey, setDuressKey] = useState<string>('');
    
    const onChangeDuressKeyInput = useCallback((e: any) => {
        setDuressKey(e.target.value)
    }, []);

    const handleAdd = useCallback(async () => {
        
    }, [duressKey]);

    return (
        <Grid className='w-100'>
            <Grid.Row centered className='p-none pb-19'>
                <Header as='h1' className='heading'>Add Duress Key</Header>
            </Grid.Row>

            <Grid.Row centered>
                <Form.Group>
                    <label>Duress Key</label>
                    <Form.Input width={8} type='password' name='duresskey' value={duressKey}
                        onChange={onChangeDuressKeyInput}/>
                </Form.Group>
            </Grid.Row>
            <Grid.Row centered>
                <Button className='primary-button pwd-confirm' onClick={e => handleAdd()}>Add</Button>
            </Grid.Row>
        </Grid >
    )
}