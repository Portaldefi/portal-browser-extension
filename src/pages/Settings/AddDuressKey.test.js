import AddDuressKey from './AddDuressKey';
import { create, act } from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Button, Form } from 'semantic-ui-react';

describe('AddDuressKeyComponent', () => {
  it('should work', () => {
    let tree;
    act(() => {
      tree = create(
        <BrowserRouter>
          <AddDuressKey />
        </BrowserRouter>
      );
    });

    expect(tree).toMatchSnapshot();
  });
  it('functions should work', async () => {
    let tree;
    act(() => {
      tree = create(
        <BrowserRouter>
          <AddDuressKey />
        </BrowserRouter>
      );
    });

    const button = tree.root.findByType(Button);
    await act(() => button.props.onClick());
    expect(tree).toMatchSnapshot();

    const input = tree.root.findByType(Form.Input);
    act(() => input.props.onChange({
      target: {
        value: 'test'
      }
    }));
    expect(tree).toMatchSnapshot();
  });
  afterAll(() => jest.resetModules());
});