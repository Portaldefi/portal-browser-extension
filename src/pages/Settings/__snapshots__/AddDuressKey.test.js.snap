// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddDuressKeyComponent functions should work 1`] = `
<div
  className="ui grid w-100"
>
  <div
    className="centered row p-none pb-19"
  >
    <h1
      className="ui header heading"
    >
      Add Duress Key
    </h1>
  </div>
  <div
    className="centered row"
  >
    <div
      className="fields"
    >
      <label>
        Duress Key
      </label>
      <div
        className="eight wide field"
      >
        <div
          className="ui input"
        >
          <input
            aria-describedby={null}
            name="duresskey"
            onChange={[Function]}
            type="password"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="centered row"
  >
    <button
      className="ui button primary-button pwd-confirm"
      onClick={[Function]}
    >
      Add
    </button>
  </div>
</div>
`;

exports[`AddDuressKeyComponent functions should work 2`] = `
<div
  className="ui grid w-100"
>
  <div
    className="centered row p-none pb-19"
  >
    <h1
      className="ui header heading"
    >
      Add Duress Key
    </h1>
  </div>
  <div
    className="centered row"
  >
    <div
      className="fields"
    >
      <label>
        Duress Key
      </label>
      <div
        className="eight wide field"
      >
        <div
          className="ui input"
        >
          <input
            aria-describedby={null}
            name="duresskey"
            onChange={[Function]}
            type="password"
            value="test"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="centered row"
  >
    <button
      className="ui button primary-button pwd-confirm"
      onClick={[Function]}
    >
      Add
    </button>
  </div>
</div>
`;

exports[`AddDuressKeyComponent should work 1`] = `
<div
  className="ui grid w-100"
>
  <div
    className="centered row p-none pb-19"
  >
    <h1
      className="ui header heading"
    >
      Add Duress Key
    </h1>
  </div>
  <div
    className="centered row"
  >
    <div
      className="fields"
    >
      <label>
        Duress Key
      </label>
      <div
        className="eight wide field"
      >
        <div
          className="ui input"
        >
          <input
            aria-describedby={null}
            name="duresskey"
            onChange={[Function]}
            type="password"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="centered row"
  >
    <button
      className="ui button primary-button pwd-confirm"
      onClick={[Function]}
    >
      Add
    </button>
  </div>
</div>
`;
